### PR TITLE
fix: preserve footnote marker punctuation adjacency

### PR DESCRIPTION
**Pull request by jan-kubica**
## Summary

- keep reconstructed footnote numbers adjacent to authored closing and terminal punctuation
- preserve a separator before ordinary prose, opening punctuation, quotes, dashes, and connectors
- resolve adjacency from the first paint-bearing run across empty text, whitespace, fields, tabs, images, math, and cached page boundaries

## Visual validation

A multi-page legal document comparison eliminated its footnote-related pagination divergence. Raster similarity improved from 0.947202 to 0.947323 and differing pixels fell by 647, while the broader document geometry remained unchanged.

## Validation

- 26 focused footnote-layout tests pass with 60 assertions
- focused lint, formatting, and diff hygiene pass
- broad validation runs in CI

CC on behalf of jan-kubica

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved footnote number spacing when followed by authored punctuation.
  * Prevented unwanted spaces before closing punctuation and punctuation in international formats.
  * Preserved appropriate separators before opening or non-attaching punctuation.
  * Improved spacing detection across visible content, including line breaks, fields, images, and mathematical content.

* **Tests**
  * Added coverage for footnote spacing and punctuation edge cases.

### CODE REVIEW

File: packages/core/src/layout-bridge/convert/footnoteLayout-table.test.ts [hunk 1]
```diff
@@ -389,8 +428,123 @@ describe("footnote layout", () => {
...
+    {
+      name: "split authored whitespace",
+      runs: [
+        { kind: "text", text: "" },
+        { kind: "text", text: " " },
+        { kind: "text", text: "Footnote text" },
+      ],
+      expected: "8 Footnote text",
+    },
+  ] satisfies { name: string; runs: Run[]; expected: string }[])(
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/core/src/layout-bridge/convert/footnoteLayout-table.test.ts:504]: Verified against the current source: each test.each table contains exactly one satisfies expression. The focused test module parses and passes all 26 tests (60 assertions), so this finding does not require a code change. The two valid style suggestions from the review summary were applied in c108d6db.\n\nCC on behalf of jan-kubica
